WooCommerce/Classes/POS/Presentation/Barcode Scanning/BarcodeScannerContainer.swift

```diff
@@ -70,4 +70,50 @@ final class GameControllerBarcodeScannerHostingController: UIHostingController<E
     deinit {
         gameControllerBarcodeObserver = nil
     }
+
+    // MARK: - VoiceOver Support
+    /// Barcode scanner  input is not received through GameController framework keyChangeHandler if VoiceOver is enabled.
+
+    override var canBecomeFirstResponder: Bool {
+        // Only become first responder when VoiceOver is running as fallback for GameController limitations
+        return UIAccessibility.isVoiceOverRunning
+    }
+
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        guard UIAccessibility.isVoiceOverRunning else { return }
+
+        becomeFirstResponder()
+    }
+
+    override func viewDidDisappear(_ animated: Bool) {
+        super.viewDidDisappear(animated)
+
+        guard UIAccessibility.isVoiceOverRunning else { return }
+
+        resignFirstResponder()
+    }
+
+    override func pressesBegan(_ presses: Set<UIPress>, with event: UIPressesEvent?) {
+        // Don't call super to prevent system from hiding software keyboard
```

Comment:
We should probably call super if VoiceOver isn't running. Perhaps it won't matter if we didn't take first responder...

Reply:
I tested it a few times, at least on my device, if I call super, even if VoiceOver is disabled, it prevents the software keyboard from showing up. Therefore, I left it at that.